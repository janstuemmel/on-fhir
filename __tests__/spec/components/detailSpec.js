jest.mock('react-native-apple-healthkit', () => {
  return {
    getWeightSamples: jest.fn()
  }
});

import React from 'react';

import { shallow, mount } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';

import HealthKit from 'react-native-apple-healthkit';

import Detail from '../../../app/components/detail';

const TEST_SAMPLES = [
  { value: '123', startDate: new Date(2017,1,1).toISOString() },
  { value: '321', startDate: new Date(2017,1,1).toISOString() },
];

describe('DetailSpec', () => {

  // afterEach(() => {
  //   jest.resetModules();
  //   jest.resetAllMocks();
  //   jest.resetAllMocks();
  // });

  it('should call componentDidMount', () => {

    // given
    var spy = jest.spyOn(Detail.prototype, 'componentDidMount');

    // when
    const wrapper = mount(
      <Detail
        label="Weight"
        identifier="weight"
        unit="gram"
        normalize={(val) => val}
      />
    );

    // then
    expect(spy).toHaveBeenCalled();

    // after
    spy.mockReset();
    spy.mockRestore();
  });

  it('should have called getWeightSamples', () => {

      // when
      const wrapper = mount(
        <Detail
          label="Weight"
          identifier="weight"
          unit="gram"
          normalize={(val) => val}
        />
      );

      // then
      expect(HealthKit.getWeightSamples).toHaveBeenCalledWith(expect.objectContaining({
          startDate: expect.any(String),
          unit: expect.any(String)
        }),
        expect.any(Function)
      );
  });


  it('should have state with samples', () => {

      // given
      HealthKit.getWeightSamples.mockImplementationOnce((opt, cb) => {
        cb(null, TEST_SAMPLES);
      });

      // when
      const wrapper = mount(
        <Detail
          label="Weight"
          identifier="weight"
          unit="gram"
          normalize={(val) => val}
        />
      );

      // then
      expect(wrapper.state()).toMatchObject({ samples: TEST_SAMPLES });
  });


  it('should have state with error', () => {

      // given
      HealthKit.getWeightSamples.mockImplementationOnce((opt, cb) => {
        cb(true, null);
      });

      // when
      const wrapper = mount(
        <Detail
          label="Weight"
          identifier="weight"
          unit="gram"
          normalize={(val) => val}
        />
      );

      // then
      expect(wrapper.state()).toMatchObject({ err: true });
  });


  it('should render items', () => {

      // given
      const wrapper = shallow(
        <Detail
          label="Weight"
          identifier="weight"
          unit="gram"
          normalize={(val) => val}
        />
      );

      // when
      wrapper.setState({ samples: TEST_SAMPLES, err: null, fetching: false });

      // then
      expect(enzymeToJson(wrapper)).toMatchSnapshot()
  });


  it('should render: data error', () => {

    // given
    const wrapper = shallow(
      <Detail
        label="Weight"
        identifier="weight"
        unit="gram"
        normalize={(val) => val}
      />
    );

    // when
    wrapper.setState({ samples: [], err: true, fetching: false });

    // then
    expect(enzymeToJson(wrapper)).toMatchSnapshot();
  });


  it('should render: data unavailable', () => {

    // given
    const wrapper = shallow(
      <Detail
        label="Weight"
        identifier="weight"
        unit="gram"
        normalize={(val) => val}
      />
    );

    // when
    wrapper.setState({ samples: [], err: null, fetching: false });

    // then
    expect(enzymeToJson(wrapper)).toMatchSnapshot();
  });

});
