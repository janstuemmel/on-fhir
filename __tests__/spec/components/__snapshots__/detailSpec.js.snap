// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailSpec should render items 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "column",
        "marginTop": 64,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#AD1457",
          "flex": 0.4,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 0.5,
            "flexDirection": "column",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 30,
            }
          }
        >
          Weight
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 0.5,
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Button
          name="Send to server"
          onPress={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
        />
      </View>
    </View>
    <ScrollView
      refreshControl={
        <RefreshControl
          onRefresh={[Function]}
          refreshing={false}
        />
      }
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "flex": 0.6,
        }
      }
    >
      <Item
        label="Feb 1, 2017"
        onPress={[Function]}
        value="123"
      />
      <Item
        label="Feb 1, 2017"
        onPress={[Function]}
        value="321"
      />
    </ScrollView>
  </View>
</View>
`;

exports[`DetailSpec should render: data error 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "column",
        "marginTop": 64,
      },
    ]
  }
>
  <Text>
    Error getting Data!
  </Text>
</View>
`;

exports[`DetailSpec should render: data unavailable 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "column",
        "marginTop": 64,
      },
    ]
  }
>
  <Text>
    No data available!
  </Text>
</View>
`;
